demo/demo.ts

```diff
@@ -191,43 +208,62 @@ export function toClosureJS(
 }
 
 function main(args: string[]): number {
-  const {settings, tscArgs} = loadSettingsFromArgs(args);
+  const { settings, tscArgs } = loadSettingsFromArgs(args);
   const config = loadTscConfig(tscArgs);
   if (config.errors.length) {
-    console.error(ts.formatDiagnostics(config.errors, ts.createCompilerHost(config.options)));
-    return 1;
-  }
-
-  if (config.options.module !== ts.ModuleKind.CommonJS) {
-    // This is not an upstream TypeScript diagnostic, therefore it does not go
-    // through the diagnostics array mechanism.
     console.error(
-        'tsickle converts TypeScript modules to Closure modules via CommonJS internally. ' +
-        'Set tsconfig.js "module": "commonjs"');
+      ts.formatDiagnostics(config.errors, ts.createCompilerHost(config.options))
+    );
     return 1;
   }
 
+  // if (config.options.module !== ts.ModuleKind.CommonJS) {
+  //   // This is not an upstream TypeScript diagnostic, therefore it does not go
+  //   // through the diagnostics array mechanism.
+  //   console.error(
+  //     "tsickle converts TypeScript modules to Closure modules via CommonJS internally. " +
+  //       'Set tsconfig.js "module": "commonjs"'
+  //   );
+  //   return 1;
+  // }
+
   // Run tsickle+TSC to convert inputs to Closure JS files.
   const result = toClosureJS(
-      config.options, config.fileNames, settings, (filePath: string, contents: string) => {
-        fs.mkdirSync(path.dirname(filePath), {recursive: true});
-        fs.writeFileSync(filePath, contents, {encoding: 'utf-8'});
-      });
+    config.options,
+    config.fileNames,
+    settings,
+    (filePath: string, contents: string) => {
+      fs.mkdirSync(path.dirname(filePath), { recursive: true });
+      fs.writeFileSync(filePath, contents, { encoding: "utf-8" });
+    }
+  );
   if (result.diagnostics.length) {
-    console.error(ts.formatDiagnostics(result.diagnostics, ts.createCompilerHost(config.options)));
+    console.error(
+      ts.formatDiagnostics(
+        result.diagnostics,
+        ts.createCompilerHost(config.options)
+      )
+    );
     return 1;
   }
 
   if (settings.externsPath) {
-    fs.mkdirSync(path.dirname(settings.externsPath), {recursive: true});
+    fs.mkdirSync(path.dirname(settings.externsPath), { recursive: true });
     fs.writeFileSync(
-        settings.externsPath,
-        tsickle.getGeneratedExterns(result.externs, config.options.rootDir || ''));
+      settings.externsPath,
+      tsickle.getGeneratedExterns(result.externs, config.options.rootDir || "")
+    );
   }
   return 0;
 }
 
-// CLI entry point
-if (require.main === module) {
-  process.exit(main(process.argv.splice(2)));
-}
+// // CLI entry point
+// if (require.main === module) {
+//   process.exit(main(process.argv.splice(2)));
+// }
+
+// if (import.meta.url.endsWith(process.argv[1])) {
+//   main(process.argv.splice(2));
+// }
```

Comment:
Commented this out since handling of CLI stuff is different with ESM and I just figured this would really always be run as the CLI anyways